source/is-float.d.ts

```diff
@@ -3,31 +3,41 @@ import type {Zero} from './numeric';
 /**
 Returns a boolean for whether the given number is a float, like `1.5` or `-1.5`.
 
-It returns `false` for `Infinity`.
-
 Use-case:
 - If you want to make a conditional branch based on the result of whether a number is a float or not.
 
 @example
 ```
-type Float = IsFloat<1.5>;
+type A = IsFloat<1.5>;
 //=> true
 
-type IntegerWithDecimal = IsInteger<1.0>;
-//=> false
+type B = IsFloat<-1.5>;
+//=> true
 
-type NegativeFloat = IsInteger<-1.5>;
+type C = IsFloat<1e-7;
 //=> true
 
-type Infinity_ = IsInteger<Infinity>;
+type D = IsFloat<1.0>;
+//=> false
+
+type E = IsFloat<PositiveInfinity>;
+//=> false
+
+type F = IsFloat<1.23+21>;
 //=> false
 ```
+
+@category Type Guard
+@category Numeric
 */
-export type IsFloat<T> =
-T extends number
-	? `${T}` extends `${infer _Sign extends '' | '-'}${number}.${infer Decimal extends number}`
-		? Decimal extends Zero
-			? false
-			: true
-		: false
+export type IsFloat<T> = T extends number
+	? `${T}` extends `${'' | '-'}${number}e${infer E extends '-' | '+'}${number}`
+		? E extends '-'
+			? true
+			: false
+		: `${T}` extends `${'' | '-'}${number}.${infer Decimal extends number}`
+			? Decimal extends Zero
+				? false
+				: true
```

Comment:
```suggestion
		: `${T}` extends `${'' | '-'}${number}.${number}`
			? true
```
<br>

`` `${T}` `` automatically removes unnecessary zeroes, for ex:
```ts
type T = `${5.00}`;
//   ^? type T = "5"
```